src/transformers/generation/utils.py

```diff
@@ -2264,6 +2264,7 @@ def sample(
 
         >>> # set pad_token_id to eos_token_id because GPT2 does not have a EOS token
         >>> model.config.pad_token_id = model.config.eos_token_id
+        >>> model.generation_config.pad_token_id = model.config.eos_token_id
```

Comment:
Here this was missing. Since the model's config was updated, the generation_config should probably be updated using some kind of `@property` utils? cc @gante

Reply:
That would solve the issue, but would make the two configs entangled with one another :( From [this comment](https://github.com/huggingface/transformers/pull/20804#issuecomment-1355151125) (and the following answers), the plan is to support ad hoc model config changes on a `generate` call for a few more versions, as dropping it would be too breaking, but drop the support in other uncommon use cases (like calling a generation submethod directly)